games-fps/urbanterror/urbanterror-4.3.2_p20171016.ebuild

```diff
@@ -0,0 +1,152 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="6"
+inherit flag-o-matic gnome2-utils
+
+DESCRIPTION="Hollywood tactical shooter based on the ioquake3 engine"
+HOMEPAGE="http://urbanterror.info https://github.com/mickael9/ioq3"
+
+if [[ ${PV} == "9999" ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/mickael9/ioq3.git"
+	EGIT_BRANCH="urt"
+else
+	COMMIT_ID="3e555ab1405c38556cbfcdd42eac23a4af49e633"
+	SRC_URI="https://github.com/mickael9/ioq3/archive/${COMMIT_ID}.tar.gz -> ${P}.tar.gz"
+	S="${WORKDIR}/ioq3-${COMMIT_ID}"
+	KEYWORDS="~amd64 ~x86"
+
+fi
+
+LICENSE="GPL-2"
+SLOT="0"
+IUSE="+altgamma +client +curl debug mumble openal +opus server voip vorbis"
+REQUIRED_USE=" || ( client server )"
+
+PATCHES=( "${FILESDIR}"/${PN}-4.3-respect_CFLAGS.patch )
+
+RDEPEND="
+	client? (
+		media-libs/libsdl2[X,sound,joystick,opengl,video]
+		mumble? ( media-sound/mumble )
+		openal? ( media-libs/openal )
+		opus? ( media-libs/opusfile )
+		vorbis? (
+			media-libs/libogg
+			media-libs/libvorbis
+		)
+	)
+	curl? ( net-misc/curl )
+	~games-fps/urbanterror-data-4.3.2
+	sys-libs/zlib[minizip]
+	virtual/jpeg:0
+"
+
+DEPEND="${RDEPEND}"
+
+pkg_pretend() {
+	if use client; then
+		if ! use openal && ! use opus && ! use vorbis; then
+			ewarn
+			ewarn "No sound implementation selected. Enable 'openal', 'opus' or 'vorbis' USE flag to get sound!"
+		fi
+	fi
+}
+
+src_compile() {
+	buildit() { use $1 && echo 1 || echo 0 ; }
+	nobuildit() { use $1 && echo 0 || echo 1 ; }
+
+	# Workaround for used zlib macro, wrt bug #44951
+	append-flags "-DOF=_Z_OF"
+
+	emake \
+		ARCH=$(usex amd64 "x86_64" "i386") \
+		DEFAULT_BASEDIR="/usr/share/urbanterror" \
+		BUILD_CLIENT=$(buildit client) \
+		BUILD_SERVER=$(buildit server) \
+		BUILD_BASEGAME=1 \
+		BUILD_MISSIONPACK=0 \
+		BUILD_GAME_SO=0 \
+		BUILD_GAME_QVM=0 \
+		BUILD_STANDALONE=1 \
+		SERVERBIN="Quake3-UrT-Ded" \
+		CLIENTBIN="Quake3-UrT" \
+		USE_RENDERER_DLOPEN=0 \
+		USE_YACC=0 \
+		BASEGAME="q3ut4"\
+		BASEGAME_CFLAGS="${CFLAGS}" \
+		USE_OPENAL=$(buildit openal) \
+		USE_OPENAL_DLOPEN=$(buildit openal) \
+		USE_CURL=$(buildit curl) \
+		USE_CURL_DLOPEN=$(buildit curl) \
+		USE_CODEC_VORBIS=$(buildit vorbis) \
+		USE_CODEC_OPUS=$(buildit opus) \
+		USE_MUMBLE=$(buildit mumble) \
+		USE_VOIP=$(buildit voip) \
+		USE_INTERNAL_LIBS=0 \
+		USE_LOCAL_HEADERS=0 \
+		USE_ALTGAMMA=$(buildit altgamma)
+}
+
+src_install() {
+	local my_arch=$(usex amd64 "x86_64" "i386")
```

Comment:
Define `my_arch` globally and use it in both `src_install` and at https://github.com/gentoo/gentoo/pull/6050/files#diff-41ab06224e69e322727523fb726ab7e0R65

Reply:
This doesn’t work in global scope.

Reply:
You're right, of course